sources/packages/web/src/components/common/students/StudentFileUploads.vue

```diff
@@ -0,0 +1,166 @@
+                  <v-icon icon="fa:far fa-file-alt" size="20"></v-icon
+                ></span>
+                <span>{{ slotProps.data.fileName }}</span>
+              </div>
+              <div v-else>
+                <span class="mr-4">
+                  <v-icon icon="fa:far fa-file-alt" size="20"></v-icon
+                ></span>
+                <span>{{ slotProps.data.fileName }}</span>
+              </div>
+            </template></Column
+          >
+        </DataTable>
+      </toggle-content>
+    </content-group>
+    <formio-modal-dialog
+      ref="fileUploadModal"
+      title="Upload file"
+      :formData="initialData"
+      formName="uploadStudentDocumentsAEST"
+    >
+      <template #actions="{ cancel, submit }">
+        <v-row class="m-0 p-0">
+          <v-btn color="primary" variant="outlined" @click="cancel"
+            >Cancel</v-btn
+          >
+          <check-permission-role :role="Role.StudentUploadFile">
+            <template #="{ notAllowed }">
+              <v-btn
+                class="float-right"
+                @click="submit"
+                color="primary"
+                variant="elevated"
+                :disabled="notAllowed"
+                >Upload now</v-btn
+              >
+            </template>
+          </check-permission-role>
+        </v-row>
+      </template>
+    </formio-modal-dialog>
+  </body-header-container>
+</template>
+
+<script lang="ts">
+import { onMounted, ref, defineComponent } from "vue";
+import { DEFAULT_PAGE_LIMIT, FormIOForm, PAGINATION_LIST, Role } from "@/types";
+import { StudentService } from "@/services/StudentService";
+import { useFormatters, useFileUtils, ModalDialog } from "@/composables";
+import FormioModalDialog from "@/components/generic/FormioModalDialog.vue";
+import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
+import { StudentUploadFileAPIOutDTO } from "@/services/http/dto/Student.dto";
+
+export default defineComponent({
+  components: {
+    CheckPermissionRole,
+    FormioModalDialog,
+  },
+  props: {
+    studentId: {
+      type: Number,
+      required: true,
+    },
+    canUploadFiles: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+    canDownloadFiles: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+  },
+  emits: ["uploadFile"],
+  setup(props, context) {
+    const studentFileUploads = ref([] as StudentUploadFileAPIOutDTO[]);
+    const fileUploadModal = ref({} as ModalDialog<FormIOForm | boolean>);
+    const { dateOnlyLongString, emptyStringFiller } = useFormatters();
+    const fileUtils = useFileUtils();
+    const initialData = ref({ studentId: props.studentId });
+
+    const loadStudentFileUploads = async () => {
+      studentFileUploads.value =
+        await StudentService.shared.getStudentFileDetails(props.studentId);
+    };
+
+    const uploadFile = () => {
+      context.emit("uploadFile", fileUploadModal, () =>
```

Comment:
I didn't follow why are we emitting back to parent here?

I technically what happen between here and parent, but are we going to follow this approach moving forward?

Reply:
I have updated this part of the code as suggested by @andrewsignori-aot

Instead of passing the entire `fileUploadModal` as argument, only `associatedFiles` array is being passed to the emitted event handler. We are emitting `uploadFile` to the AEST specific file upload component to allow for the API call `saveAESTUploadedFilesToStudent` to be made by the consumer since it is a consumer specific operation.

Reply:
@dheepak-aot considering also the last explanation given by @sh16011993 is there anything here to be considered as a new approach? 
If the point was about the modal as a parameter, that was exactly what I asked to be changed but besides that, is there anything else?

Reply:
No I am good.